angular/projects/spark-angular/src/lib/components/sprk-toggle/sprk-toggle.component.ts

```diff
@@ -121,12 +128,46 @@ export class SprkToggleComponent implements OnInit {
   getClasses(): string {
     const classArray: string[] = [
       this.titleFontClass,
-      'sprk-u-TextCrop--none',
+      'sprk-c-Toggle__trigger'
     ];
     return classArray.join(' ');
   }
 
+  /**
+   * @ignore
+   */
+  generateAriaControls(triggerElement, contentElement): void {
+    const triggerAriaControls = triggerElement.getAttribute('aria-controls');
+    let contentId = contentElement.getAttribute('id');
+
+    // Warn if aria-controls exists but the id does not
+    if (triggerAriaControls && !contentId) {
+      console.warn(`Spark Design System Warning - The component with aria-controls="${triggerAriaControls}" expects a matching id on the content element.`);
+      return;
+    }
+
+    // Warn if aria-controls and id both exist but don't match
+    if (contentId && triggerAriaControls && contentId !== triggerAriaControls) {
+      console.warn(`Spark Design System Warning - The value of aria-controls ("${triggerAriaControls}") should match the id of the content element ("${contentId}").`);
+      return;
+    }
+
+    // If we don't have a valid id, generate one with lodash
+    if (!contentId) {
+      contentId = uniqueId(`sprk_toggle_content_`);
+      contentElement.setAttribute('id', contentId);
+    }
+
+    // set the value of aria-controls
+    triggerElement.setAttribute('aria-controls', contentId);
+  }
+
   ngOnInit() {
     this.toggleState();
+
+    // Get the toggle's trigger and content elements
+    const toggleTrigger = this.ref.nativeElement.querySelector('[data-sprk-toggle="trigger"]');
```

Comment:
We should be setting attrs, adding classes etc with renderer2 (https://angular.io/api/core/Renderer2). And referencing elements inside the main component with viewchild https://www.angularjswiki.com/angular/how-to-select-an-element-in-angular-component-template/